Sources_v3/StreamChat/ChatClient.swift

```diff
@@ -259,43 +272,45 @@ public class _ChatClient<ExtraData: ExtraDataTypes> {
     ///
     init(
         config: ChatClientConfig,
+        tokenProvider: _TokenProvider<ExtraData>,
         workerBuilders: [WorkerBuilder],
         eventWorkerBuilders: [EventWorkerBuilder],
-        environment: Environment
+        environment: Environment,
+        completion: ((Error?) -> Void)? = nil
     ) {
         self.config = config
+        self.tokenProvider = tokenProvider
         self.environment = environment
         self.workerBuilders = workerBuilders
         self.eventWorkerBuilders = eventWorkerBuilders
-        
-        createBackgroundWorkers()
+
+        currentUserId = fetchCurrentUserIdFromDatabase()
         createWebSocketClient()
+        createAPIClient()
```

Comment:
Do we need to initialize API and WS clients in `init`?

Reply:
no really, fixed and updated tests in cc4bab1787e818d3a2cfcfe3613f6f9f093a0e56